auth-method-oauth2.js

```diff
@@ -1854,7 +1854,8 @@ class AuthMethodOauth2 extends AmfHelperMixin(AuthMethodBase) {
     } = this;
 
     const customGrantRequired = !isCustomGrant;
-    const clientIdRequired = ['client_credentials', 'password'].indexOf(grantType) !== -1 ? false : customGrantRequired;
+    const clientIdRequired = ['password'].indexOf(grantType) !== -1 ? false : customGrantRequired;
```

Comment:
This is not true. The spec says that this is not required for `client_credentials`. Please, remove that.

Reply:
From OAuth2 spec: https://tools.ietf.org/html/rfc6749#section-3.2.1
> A client MAY use the "client_id" request parameter to identify itself
   when sending requests to the token endpoint.

Reply:
Reverted

Reply:
TYVM!